src/cloudevents.ts

```diff
@@ -34,22 +34,89 @@ export function isBinaryCloudEvent(req: express.Request): boolean {
   );
 }
 
+/**
+ * Checks whether the incoming request is a CloudEvents event in structured content
+ * mode. This is verified by checking the presence of required body fields.
+ *
+ * @link https://github.com/cloudevents/spec/blob/master/http-protocol-binding.md#3-http-message-mapping
+ *
+ * @param req Express request object.
+ * @return True if the request is a CloudEvents event in structured content mode,
+ *     false otherwise.
+ */
+export function isStructuredCloudEvent(req: express.Request): boolean {
+  return !!(
+    req.body.specversion &&
+    req.body.type &&
+    req.body.source &&
+    req.body.id
+  );
+}
+
+/**
+ * Returns true if the request is a CloudEvent.
+ * @param req Express request object.
+ * @returns True if the request is a CloudEvent.
+ */
+export function isCloudEvent(req: express.Request): boolean {
+  return isBinaryCloudEvent(req) || isStructuredCloudEvent(req);
+}
+
+/**
+ * Converts a request to a structured CloudEvent object. Handles:
+ * - Binary CloudEvents (CloudEvent with "ce-" HTTP headers)
+ * - Structured CloudEvents (CloudEvent with event in req.body, no-op)
+ * @param req Express request object.
+ * @returns A structured CloudEvent object.
+ */
+export function convertRequestToStructuredCE(
+  req: express.Request
+): CloudEventsContext {
+  if (isBinaryCloudEvent(req)) {
+    // Binary event.
+    // Create an object with the CloudEvent properties.
+    const ce: CloudEventsContext = {
+      specversion: req.header('ce-specversion'),
+      type: req.header('ce-type'),
+      source: req.header('ce-source'),
+      subject: req.header('ce-subject'),
+      id: req.header('ce-id'),
+      time: req.header('ce-time'),
+      datacontenttype: req.header('ce-datacontenttype'),
+      data: req.body,
+    };
+    // Remove undefined keys from CE object
+    Object.keys(ce).forEach((key: string) =>
+      ce[key] === undefined ? delete ce[key] : {}
+    );
+    return ce;
+  } else {
+    // Structured event. Just return the event, which is stored in the HTTP body.
+    return req.body;
+  }
+}
+
 /**
  * Returns a CloudEvents context from the given CloudEvents request. Context
  * attributes are retrieved from request headers.
  *
  * @param req Express request object.
  * @return CloudEvents context.
  */
-export function getBinaryCloudEventContext(
+export function getStructuredCloudEventContext(
```

Comment:
I am a bit confused about how this function is different from `convertRequestToStructuredCE`